scripts/process/process-hydrograph.R

```diff
@@ -1,82 +1,79 @@
+                               rawData$Flow, 
+                               n = nrow(rawData)/viz[["full-desample"]]))
+
+  yearRange <- c(minYear, maxYear)
   
-  # format axis labels
-  labels <- seq(1890, 2016, by=10)
-  labels_dates <- as.Date(paste0(labels, "-01-01"))
-  dayssince <- labels_dates - as.Date("1889-01-01")
-  at <- dayssince*pixelDay
+  yMax <- 1.2*max(rawData$x, na.rm = TRUE)
   
   total_svg <- svglite::xmlSVG({
 
     par(omi=c(0,0,0,0), mai=c(0.5,0.75,0,0),
         las=1, xaxs = "i",mgp=c(2.5,0.25,0))
     plot(1, type='n', axes=F, ann=F, xaxt="n",
-         xlim=c(0, pixelWidth), ylim=c(-50, pixelHeight))
-    axis(side=1, at=at, labels=labels, cex.axis=0.6)
-  }, height=pixelHeight/72, width=pixelWidth/72)
+         xlim=yearRange, 
+         ylim=c(0, yMax))
+    axis(side=1, cex.axis=0.6)
+  }, height=height, width=width)
   
-  total_svg <- clean_up_svg(total_svg, viz)
+  r <- xml_find_all(total_svg, '//*[local-name()="rect"]')
+  defs <- xml_find_all(total_svg, '//*[local-name()="defs"]')
+  xml_remove(defs)
+  
+  # clean up junk that svglite adds:
+  .junk <- lapply(r, xml_remove)
 
   lines <- xml_find_all(total_svg, '//*[local-name()="line"]')
-  xml_remove(lines)
-  
-  title <- xml_find_all(total_svg, '//*[local-name()="title"]')
-  xml_remove(title)
-  
-  desc <- xml_find_all(total_svg, '//*[local-name()="desc"]')
-  xml_remove(desc)
+  .junk <- xml_remove(lines)
   
   g.year_rects <- xml_add_sibling(xml_children(total_svg)[[length(xml_children(total_svg))]], 'g', id='rectYears','class'='years-rect-all')
 
-  for(yr in names(rectangles)){
+  for(yr in c(yearRange[1]:yearRange[2])){
     rect_svg_all <- svglite::xmlSVG({
-      par(omi=c(0,0,0,0), mai=c(0.5,0.75,0,0),las=1, xaxs = "i")
+      par(omi=c(0,0,0,0), mai=c(0.5,0.75,0,0),
+          las=1, xaxs = "i")
       plot(1, type="n", axes=F, ann=F, xaxt="n",
-           xlim=c(0, pixelWidth), ylim=c(0, pixelHeight))
-      rect(xleft=rectangles[[yr]][['xleft']],
-         xright=rectangles[[yr]][['xright']],
-         ybottom=0, ytop=pixelHeight, 
-         border="blue", col="blue",
-         ann=F, xaxt="n",
-         xlim=c(0, pixelWidth), ylim=c(0, pixelHeight))
-    },  height=pixelHeight/72, width=pixelWidth/72)
-  
-
-    rect_svg <- xml_children(rect_svg_all)[4]
+           xlim=yearRange, ylim=c(0, yMax))
+      
+      rect(xleft=yr,
+         xright=yr+1,
+         ybottom=0, ytop=yMax)
+    },  height=height, width=width)
+  
+    rects <- xml_find_all(rect_svg_all, '//*[local-name()="rect"]')
+    rect_svg <- rects[length(rects)]
     
-    xml_set_attr(rect_svg, 'id', yr)
+    xml_set_attr(rect_svg, 'id', paste0("b",yr))
     xml_attr(rect_svg, "style") <- NULL
     xml_attr(rect_svg, "clip-path") <- NULL
     xml_attr(rect_svg, "class") <- "years-rect"
+    xml_attr(rect_svg, "onmouseover") <- paste0("vizlab.showline('",yr,"')")
+    xml_attr(rect_svg, "onmouseout") <- paste0("vizlab.greyline('",yr,"')")
```

Comment:
I don't know, was just starting to play around and didn't realize I committed...whoops